File: Atlab/VideoReview/VideoReview.py
# -*- coding: utf-8 -*-
from qiniu import Auth, QiniuMacAuth, http
import requests, json, sys

access_key = '<ak>'
secret_key = '<sk>'
 
auth = QiniuMacAuth(access_key, secret_key)
 
url = 'http://ai.qiniuapi.com/v3/video/censor'
 
body = {
	"data": {
		"uri": "<uri>"
	},
	"params": {
        "scenes":[],
        "cut_param": {
            "interval_msecs": 5000
        }
    }
}

ops = {
	"censor": ["pulp","terror","politician"],
	"pulp": ["pulp"],
	"terror": ["terror"],
	"politician": ["politician"],
}

body["params"]["scenes"] = ops[sys.argv[1]]
body["data"]["uri"] = sys.argv[2]

ret, res = http._post_with_qiniu_mac(url, body, auth)

headers = {"code": res.status_code, "reqid": res.req_id, "xlog": res.x_log}

print json.dumps(headers, indent=4, ensure_ascii=False)
print json.dumps(ret, indent=4, ensure_ascii=False)

File: Atlab/VideoReview/VideoReview.java
package ai;

import com.qiniu.http.Client;
import com.qiniu.http.Response;
import com.qiniu.util.Auth;
import com.qiniu.util.StringMap;
 
public class VideoReview {
     
    static final String accessKey = "<ak>";
    static final String secretKey = "<sk>";
    static final String url = "http://ai.qiniuapi.com/v3/video/censor";
    static final String video_url = "https://mars-assets.qnssl.com/scene.mp4";
    
    public static void main(String[] args) {
        StringMap headers = new StringMap();
        headers.put("Authorization", qiniuToken());
        try {
            Client client = new Client();
            Response resp = client.post(url, json().getBytes(), headers, Client.JsonMime);
            System.out.println(resp.toString());
            System.out.println(resp.bodyString());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    
    static String qiniuToken() {
        Auth auth = Auth.create(accessKey, secretKey);
        String authorization = (String) auth.authorizationV2(url, "POST", json().getBytes(), Client.JsonMime).get("Authorization");
        System.out.println(authorization);
        return authorization;
    }
     
    static String json() {
        return "{\"data\": {\"uri\": \"" + video_url  + "\"},\"params\": {\"scenes\": [\"pulp\"]}}";
    }
}
